nixos/modules/services/databases/postgresql.nix

```diff
@@ -451,7 +486,47 @@ in
           PostgreSQL superuser account to use for various operations. Internal since changing
           this value would lead to breakage while setting up databases.
         '';
+      };
+
+      upgrade = mkOption {
+        type = types.submodule {
+          options = {
+            enable = mkEnableOption "Major version automatic upgrades";
+
+            enablePreviousInstallationAutodetection = lib.mkOption {
```

Comment:
I wanted to avoid accidental "upgrades", which would start new fresh DB instead of actually upgrading. 
I found it quite difficult to distinguish clean install from upgrade and missing 'current' symlink in databaseDir.

This tries to avoid the situation, where someone upgrades the major version, but haven't run the DB at least once with this feature merged and thus not having the symlink. I expect in most cases the activation script would exit early, because the symlink exists, this just handles the one edge-case

I think the assertion is too strict and can be simplified, but I wanted to have more opinions on this before doing so

naming is pain 😅